ex/app/src/main/java/com/example/ex/MainFragment.java

```diff
@@ -0,0 +1,209 @@
+        final View v = getView();
+        if (v == null) {
+            return;
+        }
+        final Button submit = (Button)v.findViewById(R.id.b_submit);
+        submit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                saveState();
+                Log.i("submission", "State is saved");
+                setupLoading();
+                runAsync();
+                Log.i("AsyncTask", "AsyncTask was started");
+            }
+        });
+    }
+
+    private void setupLoading(){
+        progressBarStatus = 0;
+        loading = 0;
+        progressBar.setVisibility(View.VISIBLE);
+    }
+
+    private void runAsync(){
+        new Thread(new Runnable(){
+            public void run(){
+                while (progressBarStatus < HUNDRED){
+                    progressBarStatus = simulateDoingSmthInteresting();
+                    try {
+                        Thread.sleep(DEFAULT_SLEEP_TIME);
+                    } catch (final InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    progressBarHandler.post(new Runnable() {
+                        public void run() {
+                            progressBar.setProgress(progressBarStatus);
+                            if (progressBarStatus == HUNDRED){
+                                show(viewModel.getState().toString());
+                                progressBar.setVisibility(View.GONE);
+                            }
+                        }
+                    });
+                }
+                try {
+                    Thread.sleep(2 * DEFAULT_SLEEP_TIME);
+                } catch (final InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+        }).start();
+    }
+
+    private int simulateDoingSmthInteresting() {
+
+        while (loading <= UPPER_BOUND) {
+            ++loading;
+
+            if (loading == UPPER_BOUND / 3) {
+                return HUNDRED / 3;
+            } else if (loading == UPPER_BOUND * 2 / 3) {
+                return HUNDRED * 2 / 3;
+            } else if (loading == UPPER_BOUND * 4 / 5) {
+                return HUNDRED * 4 / 5;
+            }
+        }
+
+        return HUNDRED;
+
+    }
+
+    private void saveState(){
+        final State state = viewModel.getState();
+        final ArrayList<Tuple> tupleArrayList = recyclerViewAdapter.getTupleArrayList();
+
+        state.setPeople(tupleArrayList.get(0).getRating());
+        state.setAircraft(tupleArrayList.get(1).getRating());
+        state.setSeat(tupleArrayList.get(2).getRating());
+        state.setCrew(tupleArrayList.get(3).getRating());
+        if (tupleArrayList.get(4).isChecked()) {
+            state.setFood(-1);
+        } else {
+            state.setFood(tupleArrayList.get(4).getRating());
+        }
+
+        state.setText(editText.getText().toString());
```

Comment:
This thing is an anti-pattern. `RecyclerView.Adapter` simply helps us to render data from `ViewModel`, but the state should be contained in the `ViewModel` and should be always fresh (latest version) there